integrations/unstructured/src/haystack_integrations/components/converters/unstructured/converter.py

```diff
@@ -64,12 +65,11 @@ def __init__(
 
         is_hosted_api = api_url == UNSTRUCTURED_HOSTED_API_URL
 
-        api_key = api_key or os.environ.get("UNSTRUCTURED_API_KEY")
         # we check whether api_key is None or an empty string
         if is_hosted_api and not api_key:
```

Comment:
This line should be changed to actually test if the `api_key` is `None`
If you follow the previous suggestion, you can use `resolve_value`.

Reply:
yes, you are right, thanks 👍🏽